app/integration/runtime/src/main/java/io/syndesis/integration/runtime/handlers/ConnectorStepHandler.java

```diff
@@ -0,0 +1,209 @@
+        final List<String> customizers = descriptor.getConnectorCustomizers();
+
+        Map<String, String> properties = CollectionsUtils.aggregate(connection.getConfiguredProperties(), step.getConfiguredProperties());
+
+        // if the option is marked as secret use property placeholder as the
+        // value is added to the integration secret.
+        properties.entrySet()
+            .stream()
+            .filter(Predicates.or(connector::isSecret, action::isSecret))
+            .forEach(e -> e.setValue(String.format("{{%s-%s.%s}}", scheme, index, e.getKey())));
+
+        //Connector/Action properties have the precedence
+        connector.getConfiguredProperties().forEach(properties::put);
+        descriptor.getConfiguredProperties().forEach(properties::put);
+
+        try {
+            final Map<String, Object> customizersOptions = new HashMap<>();
+            properties.forEach(
+                (k, v) -> {
+                    try {
+                        customizersOptions.put(k, context.resolvePropertyPlaceholders(v));
+                    } catch (Exception e) {
+                        throw new RuntimeException(e);
+                    }
+                }
+            );
+
+            for (String customizerType : customizers) {
+                ComponentProxyCustomizer customizer = resolveCustomizer(context, customizerType);
+
+                // Set the camel context if the customizer implements
+                // the CamelContextAware interface.
+                ObjectHelper.trySetCamelContext(customizer, context);
+
+                // Bind properties to the customizer
+                IntrospectionSupport.setProperties(converter, customizer, customizersOptions);
+
+                customizer.customize(component, customizersOptions);
+            }
+
+            // Set connector options with remaining properties and original values
+            // to keep placeholders and delegate resolution to the final endpoint.
+            component.setOptions(
+                properties.entrySet().stream()
+                    .filter(e -> customizersOptions.containsKey(e.getKey()))
+                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue))
+            );
+
+            // Remove component
+            context.removeComponent(component.getComponentId());
+            context.removeService(component);
+
+            // Register component
+            context.addService(component, true, true);
+            context.addComponent(component.getComponentId(), component);
+
+            if (route == null) {
+                route = builder.from(componentId);
+            } else {
+                route = route.to(componentId);
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+
+        // Handle split
+        if (descriptor.getSplit().isPresent()) {
```

Comment:
Looks similar to the code in SimpleEndpointStepHandler. Would it make sense to move this out to a common place ?

Reply:
done